lib/offset.js

```diff
@@ -99,4 +99,47 @@ Offset.prototype.fetchLatestOffsets = function (topics, cb) {
   ], cb);
 };
 
+Offset.prototype.fetchEarliestOffsets = function (topics, cb) {
+  if (!this.ready) {
+    setTimeout(function () {
+      this.fetchEarliestOffsets(topics, cb);
+    }.bind(this), 100);
+    return;
+  }
+
+  var client = this.client;
+  async.waterfall([
+    function (callback) {
+      client.loadMetadataForTopics(topics, callback);
+    },
+    function (topicsMetaData, callback) {
+      var payloads = [];
+      var metaDatas = topicsMetaData[1].metadata;
+      Object.keys(metaDatas).forEach(function (topicName) {
+        var topic = metaDatas[topicName];
+        Object.keys(topic).forEach(function (partition) {
+          payloads.push({
+            topic: topicName,
+            partition: partition,
+            time: -2
```

Comment:
Thanks for your PR! Do you mind updating with master and refactoring this since it looks like the only difference between this and the fetchLatest is this time value.